homeassistant/components/dlink/__init__.py

```diff
@@ -1 +1,39 @@
-"""The dlink component."""
+"""The D-Link Power Plug integration."""
+from __future__ import annotations
+
+from pyW215.pyW215 import SmartPlug
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import CONF_USE_LEGACY_PROTOCOL, DOMAIN
+from .data import SmartPlugData
+
+PLATFORMS = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up D-Link Power Plug from a config entry."""
+    smartplug = await hass.async_add_executor_job(
```

Comment:
Yikes, looks like it really does swallow all expected exceptions 
https://github.com/LinuxChristian/pyW215/blob/master/pyW215/pyW215.py